lib/ProductOpener/Images.pm

```diff
@@ -467,32 +483,51 @@ sub process_image_upload($$$$$$) {
 			my $filename = get_fileid(remote_addr(). '_' . $`);
 
 			my $current_product_ref = retrieve_product($code);
-			$imgid = $current_product_ref->{max_imgid} + 1;
 
-			# if for some reason the images directories were not created at product creation (it can happen if the images directory's permission / ownership are incorrect at some point)
-			# create them
+			# if for some reason the images directories were not created
+			# at product creation (it can happen if the images directory's
+			# permission / ownership are incorrect at some point),
+			# create the the directories for the product
+			make_path("$image_base_path/", { chmod => 0755 });
 
-			# Create the directories for the product
-			foreach my $current_dir  ($www_root . "/images/products") {
-				(-e "$current_dir") or mkdir($current_dir, 0755);
-				foreach my $component (split("/", $path)) {
-					$current_dir .= "/$component";
-					(-e "$current_dir") or mkdir($current_dir, 0755);
-				}
+			my $hash = Digest::SHA3->new(224)->addfile($file, 'b')->b64digest;
+			$imgid = time() . "." . $hash;
```

Comment:
The imgid is used in the id field on input elements in the product_multilingual.pl form, it may be better to use '_' instead of '.' as a separator. 

The hash base 64 digest is also very long and can containt plus signs, slashes etc. I don't think that's a good idea for file names:

rOJhUs3nd21jgPdk7ag2fTA7/Q0/AWuAcAVwuQ.rOJhUs3nd21jgPdk7ag2fTA7/Q0/AWuAcAVwuQ

It is also very long, those imgids will be in file names, in urls, in mongo etc. I think we should put something much shorter. e.g. CRC32